runtime/io.c

```diff
@@ -429,36 +502,7 @@ void caml_finalize_channel(value vchan)
     link_channel (chan);
     return;
   }
-  caml_plat_mutex_free(&chan->mutex);
-  /* TODO KC: See commented out section */
-#if 0
   if (caml_channel_mutex_free != NULL) (*caml_channel_mutex_free)(chan);
-
```

Comment:
Is there a reason for the removal of this block?

Reply:
Actually, I think it could be kept. Let me see.